src/setfit/trainer.py

```diff
@@ -39,8 +39,9 @@ class SetFitTrainer:
         model_init (`Callable[[], SetFitModel]`, *optional*):
             A function that instantiates the model to be used. If provided, each call to [`~SetFitTrainer.train`] will start
             from a new instance of the model as given by this function when a `trial` is passed.
-        metric (`str`, *optional*, defaults to `"accuracy"`):
-            The metric to use for evaluation.
+        metric (`str` or `Callable`, *optional*, defaults to `"accuracy"`):
+            The metric to use for evaluation. If a string is provided, it must be a key present in the evaluation dataset.
```

Comment:
I wasn't sure what you meant by "key", but maybe this is a bit clearer?
```suggestion
            The metric to use for evaluation. If a string is provided, we treat it as the metric name and load it with default settings.
```

Reply:
yes sorry I was referring to metric name 😅